Randomizer/Patches/IPSPatcher.py

```diff
@@ -0,0 +1,45 @@
+from pathlib import Path
+from BaseClasses.PatchBase import *
+
+class IPSPatcher(PatchBase):
+    
+    def __init__(self, file, params = None):
+        super().__init__(file, params)        
+        
+    def __GetFile(self):
+        self.IPS_Folder = Path("IPSFiles/")
+        self.PatchFile = self.IPS_Folder / self.params
+        self.IPSFile = open(self.PatchFile, "r+b")
+    
+    def __Write(self):
+        Offset = 0
+        Size = 0
+        RLE_Size = 0
+        Value = 0
+        End_of_file_Checker = bytes(b'/x00')
+        x = 0
+        while End_of_file_Checker != bytes(b'EOF'):
```

Comment:
why in a look of 2 bytes? and not larger blocks?

Reply:
EOF is a ascii string at the end of an IPS file. Here I'm just looking if the last record (3 bytes) we loaded was the terminating string, if it's not, it's an offset, and we keep looping through.

This does share the same value as record, we could just reference record here, and I could add a comment in the script explaining it more clearly.

Reply:
Went ahead and got rid of this extra variable and referenced 'record' instead. Also added a clarifying comment that were checking for a string at the end of the file.